src/bin/ion/commands/beta/generate/templates/java/class.templ

```diff
@@ -84,18 +84,27 @@ public final class {{ target_kind_name }} {
             reader.stepOut();
         {% elif abstract_data_type is object and abstract_data_type is containing("Sequence") %}
             {# Reads `Sequence` class with a single field `value` that is an `ArrayList` #}
+            {% if abstract_data_type["Sequence"].sequence_type == "list" %}
+                if(reader.getType() != IonType.LIST) {
+                    throw new IonException("Expected list, found" + reader.getType() + "while reading {{ target_kind_name }}.");
+                }
+            {% elif abstract_data_type["Sequence"].sequence_type == "sexp" %}
+                if(reader.getType() != IonType.SEXP) {
+                   throw new IonException("Expected sexpression, found" + reader.getType() + "while reading {{ target_kind_name }}.");
+                }
+            {% endif %}
```

Comment:
(🗺️ PR tour) Adds exception statements for when `sequence_type`(list or sexp) doesn't match while reading Ion data.